LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs

```diff
@@ -12,25 +12,37 @@ namespace LibreHardwareMonitor.Hardware.Motherboard.Lpc.EC
 {
     public abstract class EmbeddedController : Hardware
     {
-        private readonly List<EmbeddedControllerSensor> _sensors;
+        private readonly EmbeddedControllerSource[] _sources;
+        private readonly List<Sensor> _sensors;
+        private readonly ushort[] _registers;
+        private readonly byte[] _data;
 
         protected EmbeddedController(List<EmbeddedControllerSource> sources, ISettings settings) : base("Embedded Controller", new Identifier("lpc", "ec"), settings)
         {
+            _sources = sources.ToArray();
```

Comment:
Maybe make _sources a list instead? Seems rather pointless to allocate an array here.

Reply:
I want to highlight that the _sources length does not change after creation. Is there a better way?

Reply:
You can make _source IReadOnlyList.

Reply:
Thank you, but from my point of view that is over complicated. Besides, I pass arrays to the EC reader object, so would need to convert on each call should I choose to leave the List here.

Reply:
_sources only seems to be used in Update to access the instances, so all you need to do change ```        private readonly EmbeddedControllerSource[] _sources;``` to ```        private readonly IReadOnlyList<EmbeddedControllerSource> _sources;``` 🙂

Reply:
For whatever reason I misread _sources for _registers, sorry. Yes, will replace with IReadOnlyList then.

Reply:
Done

Reply:
All good now to merge?

Reply:
Yes, please.